src/net_defs.h

```diff
@@ -210,6 +220,7 @@ typedef struct
 
     int num_players;
     int consoleplayer;
+    int mp_things_spawn_type; // [crispy]
```

Comment:
Doesn't this change alone break multiplayer compatibility with Chocolate Doom?

Reply:
No, as far as I can tell. I launched a modified Crispy Doom server and an unmodified Chocolate Doom client, played a couple levels and didn't notice any sync problems. Since default mp_things_spawn_type equals 0 there is no consistency problems within net play. 

Of course, if a Crispy Doom server will use any mp_things_spawn_type > 0 there will be a consistency failure while using an unmodified Chocolate Doom client. However, same will happen if server will use dm3 rules.

Reply:
I moved new optional packet reads and writes to the bottom of the net methods so they wouldn't interfere with default properties.
And I checked consistency with both a Chocolate server and a Crispy client and vice versa. I didn't notice any sync problems or errors in the console.

> Doesn't this change alone break multiplayer compatibility with Chocolate Doom?

As far as I can tell, the only proper way to send multiplayer flags to a client is by using net_gamesettings_t struct and NET_WriteSettings and NET_ReadSettings functions. So changing mp_things_spawn_type is inevitable, I think. 
Unless we are to create new net_gamesettings_extended_t struct that will inherit net_gamesettings_t properties. But then we must change a lot of other methods to be able to use the new struct.

@fabiangreffrath Do you know, if there is another correct way to send multiplayer flags?

Reply:
I completely remade the feature, following your recommendations. Now this feature only:

- Adds single command line parameter (coop2)
- Sends this game option to a client using deathmatch parameter

Notes:
Using coop2 flag game doesn’t spawn any multiplayer things except for multiplayer monsters. This variant was chosen since the many people in doom community think that is the most faithful option. We don’t spawn any extra deathmatch pickups that change the balance of a coop game but we spawn all monster that were specially placed on the map for a coop game by the id Software. Anyway, if there is an interest, all mp spawn options can be brought back using deathmatch parameters bitmask values.

Since any deathmatch value greater than 0 means that it’s a deathmatch game, a lot of code changes would be made for `deathmatch = 4` to work as coop game. So instead, there is only a small modification that uses deathmatch original value in the net operations and uses `deathmatch = 0` and `coop2 = true` values for everything else.